pymc_experimental/tests/test_model_builder.py

```diff
@@ -220,6 +220,68 @@ def test_sample_posterior_predictive(fitted_model_instance, combined):
     assert np.issubdtype(pred[fitted_model_instance.output_var].dtype, np.floating)
 
 
+@pytest.mark.parametrize("extend_idata", [True, False])
+def test_sample_prior_extend_idata_param(fitted_model_instance, extend_idata):
```

Comment:
I'm concerned about mutating the `fitted_model_instance` in these tests, as it may be used in other tests. 
In general I am not a fan of fixtures that return mutable objects that we probably want to mutate further to investigate the behavior. I suggest creating a dummy model in this test and pass a copy of the fitted idata at the beginning.

Also can we parametrize the sample method to merge this and the posterior predictive tests. They seem to share most of the logic anyway.

Reply:
I see what you're saying about modifying the `fitted_model_instance` object. My thought is to make the `fitted_model_instance` fixture return a copy of the model instance so individual tests can do what they like with it without any possibility of affecting other tests. The overhead for making a copy doesn't seem to add much to the test run time.

I originally wrote the prior and posterior predictive tests in the same test, but there were so many `if` `else` branches that I decided to split the test. But then I ended up making intermediate variables to clean up the code, so it turned out the same after all, so I'm combining them again. Thanks for the suggestion.